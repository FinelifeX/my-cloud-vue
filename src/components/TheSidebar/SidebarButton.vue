<template>
  <button class="sidebar-button" @click="$emit('click')">
    <component :is="icon"></component>
    <span class="sidebar-button__text">{{ text }}</span>
  </button>
</template>

<script>
  export default {
    name: 'SidebarButton',
    props: {
      text: {
        type: String,
        required: true,
      },
      icon: {
        type: Object,
        default: null,
      },
    },
  };
</script>

<style scoped lang="scss">
  .sidebar-button {
    @apply w-full;
    @apply flex flex-row items-center;
    @apply text-spun_pearl;
    @apply transition-colors duration-150;
    @apply pt-3 pb-3 pl-4 pr-4;
    @apply rounded-lg;

    &:hover,
    &:target,
    &:focus {
      @apply text-indigo-600;
    }

    &:active {
      @apply bg-selago;
    }
  }

  .sidebar-button__text {
    @apply ml-6;
  }
</style>
