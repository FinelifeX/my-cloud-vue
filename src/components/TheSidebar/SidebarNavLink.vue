<template>
  <router-link
    class="nav-link"
    active-class="nav-link--active"
    :exact="exact"
    :to="path"
  >
    <component :is="icon"></component>
    <span class="nav-link__text">{{ title }}</span>
  </router-link>
</template>

<script>
  export default {
    name: 'SidebarNavLink',
    props: {
      exact: {
        type: Boolean,
        default: true,
        required: false,
      },
      path: {
        type: String,
        required: true,
      },
      icon: {
        type: Object,
        default: null,
      },
      title: {
        type: String,
        required: true,
      },
    },
  };
</script>

<style scoped lang="scss">
  .nav-link {
    @apply flex flex-row items-center;
    @apply text-spun_pearl;
    @apply transition-colors duration-150;
    @apply pt-3 pb-3 pl-4 pr-4;
    @apply rounded-lg;

    &:hover {
      @apply text-indigo-600;
    }

    &--active {
      @apply text-indigo-600;
      @apply bg-selago;
      @apply font-semibold;
    }
  }

  .nav-link__text {
    @apply ml-6;
  }
</style>
