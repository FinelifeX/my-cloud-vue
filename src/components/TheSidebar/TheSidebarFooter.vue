<template>
  <footer>Help and Settings</footer>
</template>

<script>
  export default {
    name: 'TheSidebarFooter',
  };
</script>

<style scoped></style>
