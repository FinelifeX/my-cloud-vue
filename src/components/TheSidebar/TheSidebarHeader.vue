<template>
  <header class="sidebar__header">
    <div class="sidebar__header__logo">
      <cloud-outlined />
    </div>
    <span class="sidebar__header__title">My Cloud</span>
  </header>
</template>

<script>
  import CloudOutlined from '@icons/CloudOutlined.vue';

  export default {
    name: 'TheHeader',
    components: { CloudOutlined },
  };
</script>

<style scoped lang="scss">
  @import '../../assets/style/fonts';

  .sidebar__header {
    @apply flex flex-row items-center;
  }

  .sidebar__header__logo {
    @apply text-white;
    @apply bg-indigo-600;
    @apply p-3;
    @apply mr-3;
    @apply rounded-xl;
  }

  .sidebar__header__title {
    @include FontOutfit;
    @apply text-xl;
    @apply font-semibold;
  }
</style>
