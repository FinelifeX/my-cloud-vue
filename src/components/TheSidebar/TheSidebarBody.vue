<template>
  <div class="sidebar__body">
    <nav class="sidebar__nav">
      <ul class="sidebar__nav__items">
        <li v-for="link in links" :key="link.key" class="sidebar__nav__item">
          <sidebar-nav-link
            :exact="link.exact"
            :icon="link.icon"
            :path="link.path"
            :title="link.title"
          />
        </li>
      </ul>
    </nav>
  </div>
</template>

<script>
  import HomeOutlined from '@icons/HomeOutlined';
  import ImageOutlined from '@icons/ImageOutlined';
  import CameraOutlined from '@icons/CameraOutlined';
  import FilesOutlined from '@icons/FilesOutlined';
  import Routes from '@router/routes';
  import SidebarNavLink from './SidebarNavLink.vue';

  const { File } = Routes;

  export default {
    name: 'TheBody',
    components: { SidebarNavLink },
    data() {
      return {
        links: [
          {
            key: File.Home.name,
            path: File.Home.path,
            title: 'Home',
            icon: HomeOutlined,
            exact: true,
          },
          {
            key: File.Images.name,
            path: File.Images.path,
            title: 'Images',
            icon: ImageOutlined,
          },
          {
            key: File.Videos.name,
            path: File.Videos.path,
            title: 'Videos',
            icon: CameraOutlined,
          },
          {
            key: File.All.name,
            path: File.All.path,
            title: 'All files',
            icon: FilesOutlined,
          },
        ],
      };
    },
  };
</script>

<style scoped lang="scss">
  .sidebar__body {
    @apply flex-1;
    @apply pt-8;
  }

  .sidebar__nav {
  }

  .sidebar__nav__items {
  }

  .sidebar__nav__item {
    &:not(:first-child) {
      @apply mt-3;
    }

    &:not(:last-child) {
      @apply mb-3;
    }
  }
</style>
