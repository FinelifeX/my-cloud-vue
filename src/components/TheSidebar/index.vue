<template>
  <aside class="sidebar">
    <div class="sidebar__top">
      <the-sidebar-header />
      <the-sidebar-body />
    </div>
    <the-sidebar-footer />
  </aside>
</template>

<script>
  import TheSidebarHeader from './TheSidebarHeader.vue';
  import TheSidebarBody from './TheSidebarBody.vue';
  import TheSidebarFooter from './TheSidebarFooter.vue';

  export default {
    name: 'TheSidebar',
    components: { TheSidebarHeader, TheSidebarBody, TheSidebarFooter },
  };
</script>

<style scoped lang="scss">
  .sidebar {
    @apply h-full;
    @apply p-5;
    @apply flex flex-col justify-between;
  }
</style>
