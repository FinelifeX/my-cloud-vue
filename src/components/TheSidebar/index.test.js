import { render } from '@testing-library/vue';
import TheSidebar from './index.vue';

describe('TheSidebar component', () => {
  it('should render correctly', () => {
    const { html } = render(TheSidebar, { routes: [] });

    expect(html()).toMatchSnapshot();
  });
});
