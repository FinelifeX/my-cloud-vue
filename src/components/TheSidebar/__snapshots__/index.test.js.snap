// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheSidebar component should render correctly 1`] = `
"<aside class=\\"sidebar\\">
  <div class=\\"sidebar__top\\">
    <header class=\\"sidebar__header\\">
      <div class=\\"sidebar__header__logo\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"-2 -5 24 24\\" width=\\"24\\" fill=\\"currentColor\\">
          <path d=\\"M5 14a5 5 0 1 1 1.561-9.751A7.002 7.002 0 0 1 20 7a7 7 0 0 1-7 7H5zm0-2h8a5 5 0 1 0-4.6-6.965l-.72 1.686-1.742-.572A3 3 0 1 0 5 12z\\"></path>
        </svg></div> <span class=\\"sidebar__header__title\\">My Cloud</span>
    </header>
    <div class=\\"sidebar__body\\">
      <nav class=\\"sidebar__nav\\">
        <ul class=\\"sidebar__nav__items\\">
          <li class=\\"sidebar__nav__item\\"><a href=\\"#/\\" aria-current=\\"page\\" class=\\"nav-link router-link-exact-active nav-link--active\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"-2 -2 24 24\\" width=\\"24\\" fill=\\"currentColor\\">
                <path d=\\"M18 18V7.132l-8-4.8-8 4.8V18h4v-2.75a4 4 0 1 1 8 0V18h4zm-6 2v-4.75a2 2 0 1 0-4 0V20H2a2 2 0 0 1-2-2V7.132a2 2 0 0 1 .971-1.715l8-4.8a2 2 0 0 1 2.058 0l8 4.8A2 2 0 0 1 20 7.132V18a2 2 0 0 1-2 2h-6z\\"></path>
              </svg> <span class=\\"nav-link__text\\">Home</span></a></li>
          <li class=\\"sidebar__nav__item\\"><a href=\\"#/images\\" class=\\"nav-link\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"-2 -4 24 24\\" width=\\"24\\" fill=\\"currentColor\\">
                <path d=\\"M18 8.503V3a1 1 0 0 0-1-1H3a1 1 0 0 0-1 1v10a1 1 0 0 0 1 1h3.504l4.39-7.322a3 3 0 0 1 4.69-.582L18 8.503zm0 2.823l-3.828-3.814a1 1 0 0 0-1.563.195L8.836 14H17a1 1 0 0 0 1-1v-1.674zM3 0h14a3 3 0 0 1 3 3v10a3 3 0 0 1-3 3H3a3 3 0 0 1-3-3V3a3 3 0 0 1 3-3zm3 9a3 3 0 1 1 0-6 3 3 0 0 1 0 6zm0-2a1 1 0 1 0 0-2 1 1 0 0 0 0 2z\\"></path>
              </svg> <span class=\\"nav-link__text\\">Images</span></a></li>
          <li class=\\"sidebar__nav__item\\"><a href=\\"#/videos\\" class=\\"nav-link\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"-2 -2.5 24 24\\" width=\\"24\\" fill=\\"currentColor\\">
                <path d=\\"M4 11a2 2 0 0 0-2 2v2a2 2 0 0 0 2 2h6a2 2 0 0 0 2-2v-2a2 2 0 0 0-2-2H4zm9.698.473l1.888-1.887A2 2 0 0 1 17 9h.586a2 2 0 0 1 2 2v6a2 2 0 0 1-2 2H17a2 2 0 0 1-1.414-.586l-1.888-1.887A4.001 4.001 0 0 1 10 19H4a4 4 0 0 1-4-4v-2a4 4 0 0 1 4-4h6a4.001 4.001 0 0 1 3.698 2.473zM17 17h.586v-6H17l-3 3 3 3zM9.259 2.379a4.5 4.5 0 1 1-.386 5.785 5 5 0 1 1 .386-5.785zM5 16a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm0-8a3 3 0 1 0 0-6 3 3 0 0 0 0 6zm7.5 0a2.5 2.5 0 1 0 0-5 2.5 2.5 0 0 0 0 5z\\"></path>
              </svg> <span class=\\"nav-link__text\\">Videos</span></a></li>
          <li class=\\"sidebar__nav__item\\"><a href=\\"#/all-files\\" class=\\"nav-link\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"-3 -2 24 24\\" width=\\"24\\" fill=\\"currentColor\\">
                <path d=\\"M6 15H2a2 2 0 0 1-2-2V2a2 2 0 0 1 2-2h8a2 2 0 0 1 2 2v3h3l3 3v10a2 2 0 0 1-2 2H8a2 2 0 0 1-2-2v-3zm0-2V7a2 2 0 0 1 2-2h2V2H2v11h4zm8.172-6H8v11h8V8.828L14.172 7z\\"></path>
              </svg> <span class=\\"nav-link__text\\">All files</span></a></li>
        </ul>
      </nav>
    </div>
  </div>
  <footer><button class=\\"sidebar-button\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"-2 -2 24 24\\" width=\\"24\\" fill=\\"currentColor\\">
        <path d=\\"M20 8.163A2.106 2.106 0 0 0 18.926 10c0 .789.433 1.476 1.074 1.837l-.717 2.406a2.105 2.105 0 0 0-2.218 3.058l-2.062 1.602A2.104 2.104 0 0 0 11.633 20l-3.29-.008a2.104 2.104 0 0 0-3.362-1.094l-2.06-1.615A2.105 2.105 0 0 0 .715 14.24L0 11.825A2.106 2.106 0 0 0 1.051 10C1.051 9.22.63 8.54 0 8.175L.715 5.76a2.105 2.105 0 0 0 2.207-3.043L4.98 1.102A2.104 2.104 0 0 0 8.342.008L11.634 0a2.104 2.104 0 0 0 3.37 1.097l2.06 1.603a2.105 2.105 0 0 0 2.218 3.058L20 8.162zM14.823 3.68c0-.063.002-.125.005-.188l-.08-.062a4.103 4.103 0 0 1-4.308-1.428l-.904.002a4.1 4.1 0 0 1-4.29 1.43l-.095.076A4.108 4.108 0 0 1 2.279 7.6a4.1 4.1 0 0 1 .772 2.399c0 .882-.28 1.715-.772 2.4a4.108 4.108 0 0 1 2.872 4.09l.096.075a4.104 4.104 0 0 1 4.289 1.43l.904.002a4.1 4.1 0 0 1 4.307-1.428l.08-.062A4.108 4.108 0 0 1 17.7 12.4a4.102 4.102 0 0 1-.773-2.4c0-.882.281-1.716.773-2.4a4.108 4.108 0 0 1-2.876-3.919zM10 14a4 4 0 1 1 0-8 4 4 0 0 1 0 8zm0-2a2 2 0 1 0 0-4 2 2 0 0 0 0 4z\\"></path>
      </svg> <span class=\\"sidebar-button__text\\">Settings</span></button> <button class=\\"sidebar-button\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"-2 -2 24 24\\" width=\\"24\\" fill=\\"currentColor\\">
        <path d=\\"M10 20C4.477 20 0 15.523 0 10S4.477 0 10 0s10 4.477 10 10-4.477 10-10 10zm0-2a8 8 0 1 0 0-16 8 8 0 0 0 0 16zm0-3a1 1 0 1 1 0-2 1 1 0 0 1 0 2zm1.276-3.218a1 1 0 0 1-1.232-1.576l.394-.308a1.5 1.5 0 1 0-1.847-2.364l-.394.308a1 1 0 1 1-1.23-1.576l.393-.308a3.5 3.5 0 1 1 4.31 5.516l-.394.308z\\"></path>
      </svg> <span class=\\"sidebar-button__text\\">About</span></button></footer>
</aside>"
`;
